private-templates-service/client/src/components/AllCards/AllCards.tsx

```diff
@@ -38,23 +48,79 @@ const mapDispatchToProps = (dispatch: any) => {
     },
     toggleView: (viewType: ViewType) => {
       dispatch(setViewToggleType(viewType));
+    },
+    getAllTags: () => {
+      dispatch(getAllTags());
+    },
+    addSelectedTag: (tag: string) => {
+      dispatch(addSelectedTag(tag))
+    },
+    removeSelectedTag: (tag: string) => {
+      dispatch(removeSelectedTag(tag))
+    },
+    clearSelectedTags: () => {
+      dispatch(clearSelectedTags());
     }
   };
 };
 interface Props extends RouteComponentProps {
-  templates: AllTemplateState;
   setPage: (currentPageTitle: string, currentPage: string) => void;
   setSearchBarVisible: (isSearchBarVisible: boolean) => void;
   toggleView: (viewType: ViewType) => void;
+  getAllTags: () => void;
+  addSelectedTag: (tag: string) => void;
+  removeSelectedTag: (tag: string) => void;
+  clearSelectedTags: () => void;
   isSearch: boolean;
+  tags: AllTagsState;
+  selectedTags: SelectedTagsState;
+  page: PageState;
 }
 
-class AllCards extends Component<Props> {
+interface State {
+  selectedTags: string[];
+}
+class AllCards extends Component<Props, State> {
   constructor(props: Props) {
     super(props);
     this.props.setPage(ALL_CARDS_TITLE, ALL_CARDS);
     this.props.setSearchBarVisible(true);
+    this.setSelectedTags();
+  }
+  
+  setSelectedTags = (): void => {
+    if(this.props.page.currentPage === "Template" && this.props.page.previousPage === ALL_CARDS) {
```

Comment:
is it intended that tags are selected only if Template page is reached from ALL_CARDS?

Reply:
It's other way around. When we look at `currentPage` in constructor it actually will be `previousPage` since update from `setPage(ALL_CARDS..)` is not in the current props. So, when I check for `currentPage === "Template"` I'm actually asking if previous page was `Template` so I can restore state of selected templates.